workbench-core/datasets/src/ddbDataSetMetadataPlugin.ts

```diff
@@ -124,6 +127,21 @@ export class DdbDataSetMetadataPlugin implements DataSetMetadataPlugin {
     return endPointParam;
   }
 
+  public async listStorageLocations(): Promise<StorageLocation[]> {
+    const datasets = await this.listDataSets();
```

Comment:
this seems rather inefficient to query for all datasets. Keep in mind that if the return data limit exceeds 1MB, this will be missing other datasets. Revise listDataSets as it seems like it doesn't account for that. The other limitation to note here is that you can have m storage location and n amount of datasets and the time complexity grows with datasets -> O(n), not with amount of storage locations. So you will have situations where you have one storage location and 1000 datasets and you will have to look through 1000 items even though there is one storage location.

Reply:
Adding another DDB item to represent a `StorageLocation` is a good idea, but will require a larger refactor than the scope of this PR. I will make a ticket for the refactor and handle it in another PR.